liblangutil/SourceLocation.h

```diff
@@ -49,6 +49,23 @@ struct SourceLocation
 
 	bool isEmpty() const { return start == -1 && end == -1; }
 
+	/// @returns the smallest SourceLocation that contains both @param _a and @param _b.
```

Comment:
Assumes that the sources are equal.

Reply:
Well, it does work, if for ``{-1,-1,nullptr}`` as either one of the arguments... do you think we need to document that further? To me it seems obvious that this doesn't make sense two different sources...

Reply:
Yes, it should be documented.

Reply:
Ok - I tried to be precise.